crates/next-api/src/module_graph.rs

```diff
@@ -387,6 +396,73 @@ impl SingleModuleGraph {
 
         Ok(())
     }
+
+    /// Traverses all reachable edges in reverse topological order (from leaves to the entry), the
+    /// preorder visitor can be used to forward state down the graph, and to skip subgraphs
+    ///
+    /// Target nodes can be revisited (once per incoming edge).
+    /// Edges are traversed in normal order, so should correspond to reference order.
+    pub fn traverse_edges_from_entry_reverse_topological<'a, S>(
```

Comment:
Does it make sense to call it *reverse* topological when you have a preorder and postorder functor. It's basically offering both, so I would remove the `reverse` from the name